src/suricata-common.h

```diff
@@ -489,13 +489,17 @@ typedef enum {
 } LoggerId;
 
 #include "util-optimize.h"
+#ifndef SURICATA_PLUGIN
 #include <htp/htp.h>
+#endif
 #include "threads.h"
 #include "tm-threads-common.h"
 #include "util-debug.h"
 #include "util-error.h"
 #include "util-mem.h"
+#ifndef SURICATA_PLUGIN
```

Comment:
this looks hacky. Why is this needed?

Reply:
Likewise, this is causing an include problem when used from the plugins.  Just some header cleanup required still.  Right now I am sharing "suricata-plugin.h" between plugins and internal code.  But we're probably better off exposing what needs to be exposed to plugins in "suricata-plugin.h", and then an internal one for Suricata proper, with some shared stuff in its own header.  Should clean this up.

Reply:
Took a stab at fixing this without much luck.  For example, the pf-ring capture plugin needs to pull in decode.h, which pulls in a ton of stuff, re-pulling in suricata-common.h and what not resulting in "sechash.h" being pulled in which in many cases requires some paths setup. So hopefully this is temporary, until we untangle some of the includes, and/or provide some pkg-config like tool that will output compiler flags to successfully compile the plugin.